fastparquet/api.py

```diff
@@ -391,10 +392,94 @@ def remove_row_groups(self, rgs, write_fmd:bool = True,
         except IOError:
             pass
         self._set_attrs()
-
         if write_fmd:
             self._write_common_metadata(open_with)
 
+    def append_as_row_groups(self, data, row_group_offsets=None,
```

Comment:
Appends *or* overwrites

Reply:
If it is acceptable, I would propose to keep the 'append' only, having in mind we can set the `append` parameter to either `True`, `False` or `overwrite`.

Reply:
I mean, the process invoked by this method does not match its name. People would be surprised to call `append_as_row_groups` and loose some data whatever other arguments they may be passing.

Reply:
Ok,
I can propose `write_as_row_groups()` maybe?
(I am really not sure about a `append_as_or_overwrite_row_groups`, but this is certainly the farther we can go into introducing what it can do)

Maybe `set_as_row_groups` is also fitting.

Reply:
`write_row_group` ?

By the way, is "insert" possible (i.e., shuffling the other row-groups and filenames as necessary)?

Reply:
Ok for me for `write_row_group` but maybe with a `s`? (`write_row_groups`)

I don't understand the 2nd part of the comment.
Without fully understanding it, I would say 'it depends'. It depends on the mode retained in 'append' just as in current case:
- if append is True, row groups are appended
- if append is 'overwrite', ... well, it is not exactly insertion,  but replacement. This is a specific case with partitions.

Insertion of row groups is 'mimicked' in the procedure I have discussed below.
So, I would say 'yes', row groups can be shuffled thanks to `sorted`.
If I am not answering correctly, and if the discussion is related to to the 'update' procedure detailed below, I would propose to go on the discussion in the corresponding section below if this makes sense.
I should be able to answer other points raised on Wednesday evening.

Reply:
Renaming to `write_row_groups()` done.